cmd/nerdctl/exec.go

```diff
@@ -208,11 +210,24 @@ func execActionWithContainer(ctx context.Context, cmd *cobra.Command, args []str
 	return nil
 }
 
-func generateExecProcessSpec(ctx context.Context, cmd *cobra.Command, args []string, container containerd.Container) (*specs.Process, error) {
+func generateExecProcessSpec(ctx context.Context, cmd *cobra.Command, args []string, container containerd.Container, client *containerd.Client) (*specs.Process, error) {
 	spec, err := container.Spec(ctx)
 	if err != nil {
 		return nil, err
 	}
+	user, err := cmd.Flags().GetString("user")
+	if err != nil {
+		return nil, err
+	}
+	if user != "" {
+		c, err := container.Info(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if err := oci.WithUser(user)(ctx, client, &c, spec); err != nil {
```

Comment:
We might need `withResetAdditionalGIDs(), oci.WithAdditionalGIDs(user)` too https://github.com/containerd/nerdctl/blob/c3a5ce90aac4fdb22c455b54671d1f897dc2b9e0/cmd/nerdctl/run_user.go#L34